lib/api/objectPutRetention.js

```diff
@@ -78,6 +79,13 @@ function objectPutRetention(authInfo, request, log, callback) {
             parseRetentionXml(request.post, log,
                 (err, retentionInfo) => next(err, bucket, retentionInfo, objectMD));
         },
+        (bucket, retentionInfo, objectMD, next) => {
+            const validationError = validateObjectLockUpdate(objectMD, retentionInfo);
+            if (validationError) {
+                return next(validationError, bucket, objectMD);
```

Comment:
Would this be more appropriate as just ?
```suggestion
                return next(validationError);
```

Reply:
It's `async.waterfall`